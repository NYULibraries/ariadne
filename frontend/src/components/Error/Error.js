import PropTypes from 'prop-types';

const Error = ({ message }) => {
    return (
        <div>{message}</div>
    );
};

Error.propTypes = {
    message: PropTypes.string.isRequired
};

export default Error;
