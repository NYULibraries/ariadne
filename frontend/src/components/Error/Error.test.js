import { render } from '@testing-library/react';
import Error from './Error';

describe('Error component', () => {
    it('renders error message', () => {
        const { getByText } = render(<Error message="Oops! Something went wrong." />);
        expect(getByText('Oops! Something went wrong.')).toBeInTheDocument();
    });

    it('renders error message', () => {
        const { asFragment } = render(<Error message="Oops! Something went wrong." />);
        expect(asFragment()).toMatchSnapshot();
    });

});
