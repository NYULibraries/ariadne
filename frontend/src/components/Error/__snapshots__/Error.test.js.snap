// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders error message 1`] = `
<DocumentFragment>
  <div>
    Oops! Something went wrong.
  </div>
</DocumentFragment>
`;
